Modules/Core/Common/include/itkRGBAPixel.h

```diff
@@ -110,6 +110,18 @@ class ITK_TEMPLATE_EXPORT RGBAPixel:public FixedArray< TComponent, 4 >
 
   bool operator==(const Self & vec) const;
 
+  /** Implicit casting operator. Return an equivalent itk::Vector. */
+  template< typename TPixelOutput >
+  operator Vector< TPixelOutput, 4 >() const
+  {
+    Vector< TPixelOutput, 4 > v;
+    v[0] = static_cast< TPixelOutput >( this->operator[](0) );
+    v[1] = static_cast< TPixelOutput >( this->operator[](1) );
+    v[2] = static_cast< TPixelOutput >( this->operator[](2) );
+    v[3] = static_cast< TPixelOutput >( this->operator[](3) );
+    return v;
+  }
+
```

Comment:
The addition of the *implicit* conversion is significant. Which test/example required this addition?

Due to its significance please make it a separate commit in this topic.

I would like to discuss if there are/were alternatives to this addition.

Reply:
It is required to compile the example WatershedSegmentation1. An alternative is to use `RGBToVectorImageAdaptor` like [here](https://itk.org/Doxygen/html/WikiExamples_2Smoothing_2VectorGradientAnisotropicDiffusionImageFilter_8cxx-example.html).

Reply:
The example of using the adaptor with the CastImageFilter looks good to me, and preferable to adding implicit conversions.

As other future work, we might be able to "teach" the `CastImageFilter` and/or `ImageAlgorithm::Copy`  that these pixel types have the same data layout and block copies with out conversion.